### ISSUE TITLE
Bogus exception handling

### ISSUE THREAD

**Issue by danilaml** [worr/node-imdb-api#30]
Here was a bugs bug report, which I found out was actual error in my application, but I had now way of diagnosing that without digging around a and changing some code in `imdb.js`.

My app was throwing some (unexpected) exception in the `getReqById` callback, but for some reason the imdb catches all exceptions inside `prom`, so it didn't propagate further and was returned as an `cb(err, undefined)`, so I 1) had no way to find there this error actually occurred 2) thought it was internal imdb error.

**Comment by worr** [comment]
Can I see your code? We can discuss error-handling once I know if you're using the promise interface or the callback interface.

Regardless, it's unlikely that I'll be throwing exceptions rather than passing them through the callback or letting the user use the promise api to deal with them. `throw`ing errors in async code is generally frowned upon in the JS community. See an incredibly detailed explanation here: https://www.joyent.com/node-js/production/design/errors (see the section `Throw, Callback, or EventEmitter?`)

**Comment by danilaml** [comment]
It happens in both modes. IIRC the promise interface just gives some nonsensical error about `cb` not being a function or something, and the regular one just returns the exception as an `err` param.
Simple example:
```js
function myCb(err, data) {
    if (err)
        console.log(err.name);
    else
        error[1]
}

imdb.getById('tt0114709', myCb);
```

`myCB` here is called twice here. Once when the data is received and the second time here: https://github.com/worr/node-imdb-api/blob/master/src/imdb.ts#L226

This is unexpected since whether the exception/error in user's code  is caught inside imdb or not depends on where in the code it is thrown and it was hard to diagnose what happened.

**Comment by worr** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

I'm fine with this being revealed through the callback. Raising an exception doesn't make sense here.

I will fix the fact that the callback is getting called twice.

El 24 de noviembre de 2016 2:12:18 PM PST, Danila Malyutin <notifications@github.com> escribió:
>It happens in both modes. IIRC the promise interface just gives some
>nonsensical error about `cb` not being a function or something, and the
>regular one just returns the exception as an `err` param.
>Simple example:
>```js
>function myCb(err, data) {
>    if (err)
>        console.log(err.name);
>    else
>        error[1]
>}
>
>imdb.getById('tt0114709', myCb);
>```
>
>`myCB` here is called twice here. Once when the data is received and
>the second time here:
>https://github.com/worr/node-imdb-api/blob/master/src/imdb.ts#L226
>
>This is unexpected since whether the exception/error in user's code  is
>caught inside imdb or not depends on where in the code it is thrown and
>it was hard to diagnose what happened.

- --
Sent from my Android device with K-9 Mail. Please excuse my brevity.
-----BEGIN PGP SIGNATURE-----

iQI9BAEBCgAnIBxXaWxsaWFtIE9yciA8d2lsbEB3b3JyYmFzZS5jb20+BQJYPjh2
AAoJEOWIWopNkblkrTsQAI5TbUx8KcsFB2ocYSS23SpwlrbyWA4sERUsUN7ClI1O
iA+5CsdAwUPNr9csFtcpxMXC2ovzH7ziMIbbZyYL990Pys7g2xnfbs//PobqWGV2
DJix/YALLiN8jCgJe0ukONB7ZAKXNP/JpGVCyQzjp8wmRRIlIP2FQh69vcJxofSE
eeat16qCfwgEY5nZ5V+orTQ+9b4dBrgF9Puso8sjjGUP+OaazyzjyzZrj6VoQnOK
9bzHvSX2ZzOWddMO70Tk6XVur72GYyZ0NabnyzNLxRN/3NmOzSVDDyZaJ0WS9zZ3
5sdbPhayfBngjUYnWWZHXP/N2Y8/2Xxsj6vn00/55KuxBCqLzmpjwduNZaFyko1z
3sgpFddwUF1TuVBVnH6GZsstyCiZwoOuaxAYovDHpBLMbMkaKc4PR2zjW7UpCqsI
LwDmi8ZoHjiiMJ1PqJ/Ed60G923lwFjIYCUUHhrAGCKtCQb+cTROhBoc9Ju1Bc6P
0EEhycRrLX7Yt8mEog0/WFscLE05TGevixSP/BCJAh2TNGiOeX2hVwEuwconTo1e
fBBz3GqRGTUGtDBltZSqaG6kU4K54hWsQfzyouxKHQDQ3itEraCI83i56PyyAvE4
YDqzAyEv/ZeRLRXnIKfCXN8AsAgDD030+erp2e314EEIrqAtMT27W2rJ8EMpOwCJ
=v7uU
-----END PGP SIGNATURE-----

**Comment by worr** [comment]
This has been fixed in c0d11066f8e424a80e15c4a444fb0ab55b0c7cf4
